plugins/lsp/language-servers/nil_ls-settings.nix

```diff
@@ -1,42 +1,58 @@
 { lib, helpers }:
-# Options:
-#  - https://github.com/oxalica/nil/blob/main/docs/configuration.md?plain=1#
+# All available settings are documented here:
+# https://github.com/oxalica/nil/blob/main/docs/configuration.md
 with lib;
 {
   formatting = {
-    command = helpers.defaultNullOpts.mkListOf types.str "null" ''
-      External formatter command (with arguments).
-      It should accepts file content in stdin and print the formatted code into stdout.
-    '';
+    command = helpers.defaultNullOpts.mkListOf' {
+      type = types.str;
+      default = null;
+      description = ''
+        External formatter command (with arguments).
+        It should accepts file content in stdin and print the formatted code into stdout.
+      '';
+      example = [ "nixpkgs-fmt" ];
+    };
   };
 
   diagnostics = {
-    ignored = helpers.defaultNullOpts.mkListOf types.str "[]" ''
+    ignored = helpers.defaultNullOpts.mkListOf types.str [ ] ''
       Ignored diagnostic kinds.
       The kind identifier is a snake_cased_string usually shown together
       with the diagnostic message.
     '';
 
-    excludedFiles = helpers.defaultNullOpts.mkListOf types.str "[]" ''
-      Files to exclude from showing diagnostics. Useful for generated files.
-      It accepts an array of paths. Relative paths are joint to the workspace root.
-      Glob patterns are currently not supported.
-    '';
+    excludedFiles = helpers.defaultNullOpts.mkListOf' {
+      type = types.str;
+      default = [ ];
+      description = ''
+        Files to exclude from showing diagnostics. Useful for generated files.
+        It accepts an array of paths. Relative paths are joint to the workspace root.
+        Glob patterns are currently not supported.
+      '';
+      example = [ "Cargo.nix" ];
+    };
   };
 
   nix = {
-    binary = helpers.defaultNullOpts.mkStr "nix" ''
-      The path to the `nix` binary.
-    '';
+    binary = helpers.defaultNullOpts.mkStr' {
```

Comment:
`mkStr'` doesn't exist. I can make a PR if you think it's useful.

Reply:
Why not actually. I like the idea.